core/common/src/main/java/alluxio/heartbeat/HeartbeatScheduler.java

```diff
@@ -54,25 +58,45 @@ private HeartbeatScheduler() {} // to prevent initialization
   public static void addTimer(ScheduledTimer timer) {
     Preconditions.checkNotNull(timer);
     try (LockResource r = new LockResource(sLock)) {
+      Preconditions.checkState(!sTimers.containsKey(timer.getThreadName()),
+          "The timer for thread %s is already waiting to be scheduled", timer.getThreadName());
       sTimers.put(timer.getThreadName(), timer);
       sCondition.signalAll();
     }
   }
 
   /**
-   * @param timer a timer to remove from the scheduler
+   * Removes a timer name from the scheduler if it exists.
+   *
+   * @param name the name to remove
    */
-  public static synchronized void removeTimer(ScheduledTimer timer) {
-    Preconditions.checkNotNull(timer);
+  public static void removeTimer(String name) {
```

Comment:
I think its a little confusing that the two `removeTimer` methods have different behaviors.